### PR TITLE
#1679 Landing page information takes a minute to load with 1.3M runs

### LINKED ISSUE #1679: Landing page information takes a minute to load with 1.3M runs
## Describe the bug
The landing page statistics take a shocking 1 minute to load when there's a lot of entities and namely runs. From UX perspective this is quite unacceptable.

This query should be cached or regularly materialized - the landing page statistics - graphs & tiles should load almost immediately.

### PR DESCRIPTION
**Pull request by lokm01**
Fixes #1679 

This PR proposes how such expensive queries could be addressed before we invest into the massive data model rebuild or before we got requirements with more advanced caching needs (where products like redis could be used)

Here Spring will run the recalculation of the landing page statistics upon the start up and subsequently in 15 minute intervals. These statistics are stored in a mongo collection and the API endpoint simply reads the latest record.

### CODE REVIEW

File: menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala [hunk 1]
```diff
@@ -48,4 +59,14 @@ class LandingPageController @Autowired() (datasetRepository: DatasetMongoReposit
       todaysStats <- runsService.getTodaysRunsStatistics()
     } yield LandingPageInformation(dsCount, mtCount, schemaCount, runCount, todaysStats)
   }
+
+  @Async
+  @Scheduled(fixedDelay = 900000)
+  def scheduledTest(): CompletableFuture[_] = {
```
Comment (dk1844) [inline menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala:65]:
```suggestion
  @Scheduled(initialDelay = 1000, fixedDelay = 900000)
  def scheduledLandingPageStatsRecalc(): CompletableFuture[_] = {
```
I hope I understand the `@Scheduled` annotation correctly, but it seems to me that we are risking that in the first 15 minutes there is no landing page info cached.

**Review by dk1844 — APPROVED** [review]

**Review by dk1844 — APPROVED** [review]
(Just read the code)

File: menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala (hunk 1 above)
Comment (lokm01) [reply menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala:65]: Good point, I also wondered! :) Thanks

**Review by dk1844 — DISMISSED** [review]

File: menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala [hunk 2]
```diff
@@ -48,4 +59,14 @@ class LandingPageController @Autowired() (datasetRepository: DatasetMongoReposit
       todaysStats <- runsService.getTodaysRunsStatistics()
     } yield LandingPageInformation(dsCount, mtCount, schemaCount, runCount, todaysStats)
   }
+
+  @Async
+  @Scheduled(initialDelay = 1000, fixedDelay = 900000)
```
Comment (benedeki) [inline menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala:64]: Also, is 15 minutes refresh a good frequency?

**Review by benedeki — COMMENTED** [review]
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run

The comments are about code styling & typos.
Generally it's more a workaround then proper solution IMHO. But seems to work as intended.

File: menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala (hunk 2 above)
Comment (lokm01) [reply menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala:64]: Good question, these annotations also expect constants so I couldn't find a quick way to make them configurable

Comment (benedeki) [reply menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala:64]: As written elsewhere, I hope this is a quickfix. So let's not dwell on it for too long. 
On second thought, 15 minutes or 5 doesn't make much difference :-)

Comment (lokm01) [reply menas/src/main/scala/za/co/absa/enceladus/menas/controllers/LandingPageController.scala:63]: Hmm maybe 5 mins is more reasonable from user perspective.

**Review by benedeki — APPROVED** [review]

**Pull request merged**